src/redux/userSlice.ts

```diff
@@ -0,0 +1,27 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import type { RootState } from './store';
+
+// declaring the types for our state
+export type UserState = {
+  id: number;
+};
```

Comment:
State signature should look something like this:

```ts
export type UserState = {
  credentials: {
    id: string;
    role: string;
  };
  user: IUser
};
```

and the rest of the slice should implement this interface